### ISSUE TITLE
Feature request: add edit_parameters to toolbar for all GUI backends

### ISSUE THREAD

**Issue by BYT3M3** [matplotlib/matplotlib#13195]
Dear maintainer

So I have developed a PyQt application with matplotlib in the past and now I wanted to try it with Gtk because prettier (in my humble opinion) and probably easier on the dependencies.

However, I very much miss the "edit parameters" button on the Gtk toolbar, which is present on the Qt toolbar. I would very much like it if this button is also implemented for the Gtk backend or preferably straight into the matplotlib.backend_bases.NavigationToolbar2 class, because I also miss the "edit parameter" button when using pyplot for quickly hacking a graph together.

Thank you for matplotlib, I love it!

**Comment by tacaswell** [comment]
I agree it would be great if the gtk backends also had the 'edit parameters' button!   I do not think it can go into the base class though as to make that GUI nice, it should be implemented using the native widgets of the given toolkit.  

One of the limiting factors in this case in expertise in gtk among the current core-developers, if you are currently work on a Gtk application @BYT3M3 I am going to assume you do have expertise in gtk, would you be interested in developing (at least the start) of that window?

**Comment by BYT3M3** [comment]
@tacaswell : I don't have any gtk experience ­— let alone expertise — whatsoever. I just wanted my UI to look a little fancier and to possibly have less issues with packaging the final program, because PyQt is a PITA to package.

Anyway, I might find some time to hack together a start for the GUI, I could certainly throw a design together in glade and provide some interface. It's writing the actual methods that change everything that may be a little out of my comfort zone. I can't give any guarantees on when I'll do this, but it's good to know that effort I put in could lead to completing the feature.

**Comment by tacaswell** [comment]
> It's writing the actual methods that change everything that may be a little out of my comfort zone.

Luckily that is what people around here have expertise in!  If you have any questions about how to update artist properties please ask!

https://github.com/matplotlib/matplotlib/blob/4432fdaa7a0af4f8a22a07f1df1792b026446422/lib/matplotlib/backends/qt_editor/figureoptions.py#L178

That is the function in the Qt case that actually does the Artist update, that would probably be a good place to start (also to get an idea of what 'feature parity' would look like).

**Comment by BYT3M3** [comment]
I'm still interested in having this feature. Unfortunately, I have not found the time to implement it myself.
